common/workunit/workunit.cpp

```diff
@@ -2889,9 +2889,12 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                     namefilter.set(fv);
                 else if (subfmt==WUSFappvalue)
                 {
-                    query.append("[Application/").append(fv).append("=?~\"");
+                    const char *app = fv;
                     fv = fv + strlen(fv)+1;
-                    query.append(fv).append("\"]");
+                    query.append("[Application/").append(app);
+                    if (fv && *fv)
```

Comment:
minor: test for fv is redundant.  

The embedded null character isn't a good interface, but it looks like that's what it did before.